### PR TITLE
chore: namespace component refactor

### LINKED ISSUE #722: Namespace refactor

### PR DESCRIPTION
**Pull request by veds-g**
Fixes #722 
- modified the search bar view
-- removed the search button
-- searches when pressed 'Enter' after typing or selecting from the dropdown
- increased the cache size of previously searched namespaces from 5 to 10
- added a label "Pipelines" when listing available pipelines for the queried namespace

image

image

### CODE REVIEW

File: ui/src/components/pages/Namespace/index.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
+import { useEffect, useState } from "react";
+import { NamespaceRowContent } from "./partials/NamespaceRowContent";
+import "./style.css";
```
Comment (mshakira) [inline ui/src/components/pages/Namespace/index.tsx:3]: please move this to end of imports line

File: ui/src/components/pages/Namespace/partials/NamespaceRowContent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { Link } from "react-router-dom";
+import Box from "@mui/material/Box";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import { useNamespaceFetch } from "../../../../../utils/fetchWrappers/namespaceFetch";
+import { notifyError } from "../../../../../utils/error";
+import { useEffect } from "react";
```
Comment (mshakira) [inline ui/src/components/pages/Namespace/partials/NamespaceRowContent/index.tsx:7]: plz move it on top

File: ui/src/components/pages/Namespace/index.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  // provides namespace search based on scoped installation
+  useEffect(() => {
+    if (systemInfo && systemInfo?.namespaced) {
+      setNamespace(systemInfo?.managedNamespace);
+      setDisableSearch(true);
+    } else if (systemInfo && !systemInfo?.namespaced) {
+      setDisableSearch(false);
+
+      // set namespace value for search box
+      const curr_ns = GetStore("curr_namespace");
```
Comment (mshakira) [inline ui/src/components/pages/Namespace/index.tsx:72]: We can add curr_namespace as a global variable. In that way, when its used multiple times, its easy to manage

File: ui/src/components/pages/Namespace/index.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      arr.push({
+        label: namespace,
+        type: "Available Namespaces",
+      });
+    });
+    setNsCombined(arr);
+  }, [nsCookie, nsCluster]);
+
+  const handle = (namespaceVal: string) => {
+    SetStore("curr_namespace", namespaceVal);
```
Comment (mshakira) [inline ui/src/components/pages/Namespace/index.tsx:110]: same here

**Review by mshakira — APPROVED** [review]
LGTM

**Review by whynowy — APPROVED** [review]

**Pull request merged**